pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -351,7 +351,9 @@ func (m *vmMutator) getAdditionalGuestMemoryOverheadRatioConfig() (*settings.Add
 }
 
 func (m *vmMutator) patchAffinity(vm *kubevirtv1.VirtualMachine, patchOps types.PatchOps) (types.PatchOps, error) {
-	if vm == nil || vm.Spec.Template == nil {
+	// if vm is stopped/terminating/stopping, do not patch affinity, it makes no sense, and may block the deletion of vm
+	// e.g. the vm's nad has been deleted but vm's interface is dangling
+	if vm == nil || vm.Spec.Template == nil || vm.DeletionTimestamp != nil || vm.Status.PrintableStatus == kubevirtv1.VirtualMachineStatusStopped || vm.Status.PrintableStatus == kubevirtv1.VirtualMachineStatusStopping || vm.Status.PrintableStatus == kubevirtv1.VirtualMachineStatusTerminating {
```

Comment:
Hi @w13915984028, since #8110 is merged, we can remove `vm == nil` and `vm.DeletionTimestamp != nil` here. Thanks.

Reply:
@FrankYang0529 Thanks for your reminder, I will double check related PRs and update accordingly.